File: src/main/java/cn/skullmind/io/server/login/LoginController.java
package cn.skullmind.io.server.login;

import cn.skullmind.io.server.beans.User;
import cn.skullmind.io.server.http.Resp;
import org.apache.juli.logging.Log;
import org.springframework.web.bind.annotation.*;

import java.util.logging.Logger;

@RestController
public class LoginController {

    /***
     * 用户注册String
     * @return
     */
    @RequestMapping(value = "/user",method = RequestMethod.POST)
    public Resp<User> addUser(@RequestParam String name,@RequestParam String mobile,
                        @RequestParam long userId,@RequestParam boolean gender){
        User user = new User();
        user.setName(name);
        user.setMobile(mobile);
        user.setGender(gender);
        user.setUserId(userId);
        Resp<User> resp = new Resp<>(200,"register success",user);
        return resp;
    }

    @RequestMapping(value = "/user/{userId}",method = RequestMethod.GET)
    public Resp<User> queryUser(@PathVariable long userId){
        User user = new User();
        user.setName("xujianhua");
        user.setMobile("18512177309");
        user.setGender(false);
        user.setUserId(userId);
        Resp<User> resp = new Resp<>(200,"query success",user);
        return resp;
    }

}

File: server_mobile/src/main/java/cn/skullmind/io/mobile/controller/login/LoginController.kt
package cn.skullmind.io.mobile.controller.login

import cn.skullmind.io.mobile.beans.UserInfo
import cn.skullmind.io.mobile.http.Resp
import cn.skullmind.io.mobile.service.UserInfoServiceImp
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.web.bind.annotation.*

@RestController
class LoginController {
    @Autowired
    var service:UserInfoServiceImp?=null

    /***
     * 用户注册String
     * @return
     */
    @RequestMapping(value = ["/user"],method = [RequestMethod.POST],
            produces = ["application/json;charset=UTF-8"])
    @ResponseBody
    fun addUser(@RequestParam wechatId:String,
                @RequestParam mobile:String,
                @RequestParam password:String):Resp<UserInfo?>{

        val user = UserInfo(mobile,wechatId,password)
        val insertUser = service?.insertByUserInfo(user)
        return Resp(200,"register success",insertUser)
    }

    @RequestMapping(value = ["/user/{userId}"],method = [RequestMethod.GET])
    fun  queryUser(@PathVariable  userId:Int):Resp<UserInfo?>{
         val user = service?.findById(userId)
         return Resp(200,"query success",user)
    }

    @RequestMapping(value = ["/user/{userId}"],method = [RequestMethod.DELETE])
    fun  deleteUser(@PathVariable  userId:Int):Resp<UserInfo?>{
        service?.delete(userId)
        return Resp(200,"delete success",null)
    }

}
